cloudferrylib/os/actions/recreate_boot_image.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import files
+from cloudferrylib.utils import remote_runner
+from cloudferrylib.utils.drivers.ssh_chunks import verified_file_copy, remote_md5_sum
+from cloudferrylib.utils import utils
+import copy
+import os
+
+
+LOG = utils.get_log(__name__)
+
+
+class ReCreateBootImage(action.Action):
+
+    def run(self, images_info=None, compute_ignored_images={}, missing_images={}, **kwargs):
+        images_info = copy.deepcopy(images_info)
+        src_user = self.cfg.src.ssh_user
+        src_password = self.cfg.src.ssh_sudo_password
+        src_host = self.cfg.src.ssh_host
+        dst_user = self.cfg.dst.ssh_user
+        dst_password = self.cfg.dst.ssh_sudo_password
+        dst_host = self.cfg.dst.ssh_host
+        for vm_id in missing_images:
```

Comment:
As part of refactoring please try to follow pep8 guidelines, primarily line length (I know it can be tricky, but several lines are not fully visible even in github)